test/cbmc/include/defender_cbmc_state.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file defender_cbmc_state.h
+ * @brief Allocation and assumption utilities for the Device Defender Client
+ * library CBMC proofs.
+ */
+
+#ifndef DEFENDER_CBMC_STATE_H_
+#define DEFENDER_CBMC_STATE_H_
+
+/* Standard includes. */
+#include <stddef.h>
+
+/**
+ * @brief Proof model for malloc that can fail and return NULL.
+ *
+ * @param[in] size The size (in bytes) of memory to allocate.
+ *
+ * @return NULL or requested memory.
+ */
+void * mallocCanFail( size_t size );
```

Comment:
The current version of the starter kit runs cbmc by default with flags that allow malloc to fail (can return either NULL or a valid pointer).  So this may not be explicitly required anymore, but you may wish to retain it for documentation (eg, drive home the point that you are considering the possibility that malloc can fail).